src/plugins/interactions/src/components/effect/deal_damage.rs

```diff
@@ -1,22 +1,43 @@
 use super::Effect;
-use crate::traits::{act_on::ActOn, is_effect::IsEffect};
-use bevy::prelude::Entity;
+use crate::{traits::act_on::ActOn, InteractionsPlugin};
+use bevy::prelude::*;
 use common::{
-	components::Health,
+	attributes::health::Health,
 	effects::{deal_damage::DealDamage, EffectApplies},
+	traits::{
+		handles_effect::HandlesEffect,
+		handles_life::{ChangeLife, HandlesLife},
+	},
 };
 use std::time::Duration;
 
-impl IsEffect for Effect<DealDamage> {}
+impl<TLifecyclePlugin> HandlesEffect<DealDamage> for InteractionsPlugin<TLifecyclePlugin>
+where
+	TLifecyclePlugin: HandlesLife,
+{
+	type TTarget = Health;
 
-impl ActOn<Health> for Effect<DealDamage> {
-	fn act(&mut self, _: Entity, health: &mut Health, delta: Duration) -> EffectApplies {
+	fn effect(effect: DealDamage) -> impl Bundle {
+		Effect(effect)
+	}
+
+	fn attribute(health: Self::TTarget) -> impl Bundle {
+		TLifecyclePlugin::TLife::from(health)
+	}
+}
+
+impl<TLife> ActOn<TLife> for Effect<DealDamage>
+where
+	TLife: ChangeLife,
+{
+	fn act(&mut self, _: Entity, life: &mut TLife, delta: Duration) -> EffectApplies {
 		let Effect(DealDamage(damage, apply_method)) = *self;
 
-		health.current -= match apply_method {
-			EffectApplies::Always => damage * delta.as_secs_f32(),
-			EffectApplies::Once | EffectApplies::OncePerTarget => damage,
+		let change = match apply_method {
+			EffectApplies::Always => -damage * delta.as_secs_f32(),
+			EffectApplies::Once | EffectApplies::OncePerTarget => -damage,
 		};
+		life.change_by(change);
 
 		apply_method
 	}
```

Comment:
Maybe use in both cases the plugin as dependency?

Reply:
Nope, because the dependencies depend on the structs/traits we are trying to implement. Cannot arbitrarily change those out.